airbyte-integrations/bases/base-java/src/main/java/io/airbyte/integrations/destination/dest_state_lifecycle_manager/DestStateLifecycleManager.java

```diff
@@ -16,6 +16,9 @@
  * been committed</li>
  * <li>committed - associated records have been committed</li>
  * </ol>
+ *
+ * Since the methods for flushing and committing are not atomic, adding in a method
```

Comment:
I don't understand this comment. How are method for flushing and committing not atomic and why that means that instead of `markPendingAsFlushed` is replaced with `markPendingAsCommitted`

Reply:
Good catch, this was mostly a quick comment that I forgot to update, I'll update this but the main point was that since we're bundling flushing and committing that `markPendingAsFlushed` will be superseded by `markPendingAsCommitted`